### PR TITLE
fixed issue #2276

### PR DESCRIPTION
**Pull request by hyi**
### Pull Request Checklist: 
- [ ] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. After logging into HydroShare, edit your profile page and update your middle name, e.g., add a middle name or remove middle name, and save your profile. Then go create a resource, make sure the updated middle name from your profile edit is reflected in author/creator display on resource landing page and my resources page.

### CODE REVIEW

**Comment by hyi** [comment]
@pkdash I assigned you as the code reviewer for this PR since you implemented the party model and are best positioned to review this small code change, and yes, the code change is small so it should be an easy code review. If you don't have time, let me know and I can try to ask someone else to review it. Thanks.

**Comment by hyi** [comment]
@pkdash Just realized this code change may break some existing tests. I will fix existing test failures if any after Jenkins check finishes.

File: hs_core/hydroshare/utils.py [hunk 1]
```diff
@@ -794,7 +794,13 @@ def prepare_resource_default_metadata(resource, metadata, res_title):
 def get_party_data_from_user(user):
     party_data = {}
     user_profile = get_profile(user)
-    user_full_name = user.get_full_name()
+
+    if user_profile.middle_name:
```
Comment (sblack-usu) [inline hs_core/hydroshare/utils.py]: Shouldn't this be in the get_full_name method on the AbstractUser model?

Comment (hyi) [reply hs_core/hydroshare/utils.py]: @sblack-usu AbstractUser model is the default Django auth user model which we have no control over to change, and we are using the default Django user model which does not account for middle name at all in its user model as well as this get_full_name() function. It'd be ideal if we had our own customized user model to start with, but at this point we just have to live with whatever Django default user model provide to us.

Comment (sblack-usu) [reply hs_core/hydroshare/utils.py]: oh, I didn't realize AbstractUser existed in a django library.  I take back my comment and should have been a bit more thorough looking this over.

**Pull request merged**